File: backend/src/main/java/fr/pedalons/enums/NotificationDeliveryStatus.java
package fr.pedalons.enums;

/** Lifecycle of one notification on one out-of-app channel. */
public enum NotificationDeliveryStatus {
  /** Waiting for {@code nextAttemptAt}. */
  PENDING,
  /** Claimed by a sender tick. */
  SENDING,
  /** Handed to the provider. */
  SENT,
  /** Not sent on purpose — the recipient was deleted, or the channel was switched off since. */
  SKIPPED,
  /** Gave up after repeated failures. */
  FAILED
}

File: frontend/src/api/dto/notificationDeliveryStatus.ts
export type NotificationDeliveryStatus =
  (typeof NotificationDeliveryStatus)[keyof typeof NotificationDeliveryStatus]

export const NotificationDeliveryStatus = {
  PENDING: 'PENDING',
  SENDING: 'SENDING',
  SENT: 'SENT',
  SKIPPED: 'SKIPPED',
  FAILED: 'FAILED',
} as const
